javatests/arcs/sdk/wasm/OnCreateTest.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.sdk.wasm
+
+class OnCreateTest : AbstractOnCreateTest() {
+    var s = "Not created!"
+
+    override fun onCreate() {
+        s = "Created!"
```

Comment:
I thought the point of onCreate was to write default state to handles? If that's the case, then a more complete test would be to do that here instead. So you could call `handles.fooHandle.store("Created!")` here, and remove the `onHandleSync` stuff.

Reply:
The test that @SHeimlich is doing below is allowing her to distinguish between a boot up sequence in which onCreate() is called, vs. a sequence where onCreate() is *not* called.

If the only writing to handles happened in onCreate(), then when testing the deserialization flow and not seeing anything written, one could wonder was the particle booted at all. With the test structure like Sarah has, it is clear that the particle has booted, and that onCreate was not called (at least not before onHandleSync()).

One thing that could be improved, is ensuring that handles are ready for writing when onCreate() gets invoked. The particle could be restructured like below to achieve that:
```
class OnCreateTest : AbstractOnCreateTest() {
    var onCreateCalled = false

    override fun onCreate() {
        onCreateCalled = true;
        handles.fooHandle.store(OnCreateTest_FooHandle("Created!"))
    }

    override fun onHandleSync(handle: WasmHandle, allSynced: Boolean) {
        if (!onCreateCalled) {
          handles.fooHandle.store(OnCreateTest_FooHandle("Not created!"))
        }
    }
}
```

WDYT?

Reply:
The remaining thing that is not explicitly tested in the deserialization flow is that onCreate() never gets called (vs. before onHandleSync), but I think we can live with it.

Reply:
This didn't work initially due to a bug in the TS implementation. That is being resolved with #4797